odl/tomo/geometry/conebeam.py

```diff
@@ -0,0 +1,557 @@
+
+        src_rad = float(src_rad)
+        if src_rad <= 0:
+            raise ValueError('source circle radius {} is not positive.'
+                             ''.format(src_rad))
+        det_rad = float(det_rad)
+        if det_rad <= 0:
+            raise ValueError('detector circle radius {} is not positive.'
+                             ''.format(det_rad))
+
+        if agrid is not None:
+            if not isinstance(agrid, TensorGrid):
+                raise TypeError('angle grid {!r} is not a `TensorGrid` '
+                                'instance.'.format(agrid))
+            if not angle_intvl.contains_set(agrid):
+                raise ValueError('angular grid {} not contained in angle '
+                                 'interval {}.'.format(agrid, angle_intvl))
+
+        angle_offset = float(angle_offset)
+
+        super().__init__()
+        self._motion_params = angle_intvl
+        self._src_radius = src_rad
+        self._det_radius = det_rad
+        self._motion_grid = agrid
+        self._motion_params_offset = angle_offset
+        self._axis = axis
+
+    @property
+    def motion_params(self):
+        """Motion parameters of this geometry."""
+        return self._motion_params
+
+    @property
+    def motion_params_offset(self):
+        """Offset to motion parameters. """
+        return self._motion_params_offset
+
+    @property
+    def motion_grid(self):
+        """Sampling grid for this geometry's motion parameters."""
+        return self._motion_grid
+
+    @property
+    def angle_intvl(self):
+        """Angles (= motion parameters) of this geometry."""
+        return self._motion_params
+
+    @property
+    def angle_grid(self):
+        """Angle (= motion parameter) sampling grid of this geometry."""
+        return self._motion_grid
+
+    @property
+    def angle_offset(self):
+        """Offset to the rotation angle in the azimuthal plane. Does not
+        imply an offset in z-direction. The actual angles then reside
+        within `angle_offset` + `angle_intvl`."""
+        return self._motion_params_offset
+
+    @property
+    def src_radius(self):
+        """Source circle radius of this geometry."""
+        return self._src_radius
+
+    @property
+    def det_radius(self):
+        """Detector circle radius of this geometry."""
+        return self._det_radius
+
+    @property
+    def ndim(self):
+        """Number of dimensions of this geometry."""
+        return 3
+
+    @property
+    def axis(self):
+        """The axis of rotation.
+
+        Returns
+        -------
+        axis : `numpy.ndarray`, shape ``(3,)``
+            The rotation axis
+        """
+        axis = self._axis
+        if axis is None:
+            axis = np.array([0, 0, 1])
+        elif isinstance(axis, (int, float)):
+            ind = axis
+            axis = np.zeros(3)
+            axis[ind] = 1
+        elif len(axis) == 3:
+            axis = np.array(axis)
```

Comment:
Question: normalize here (implying a error if we get zero) or use a normalized rotation formula or nothing of that? I'm in favor of the first variant.